keras/backend/theano_backend.py

```diff
@@ -357,8 +357,11 @@ def __call__(self, inputs):
         return self.function(*inputs)
 
 
-def function(inputs, outputs, updates=[]):
-    return Function(inputs, outputs, updates=updates)
+def function(inputs, outputs, updates=[], debug_nan=False):
+    mode = None
+    if _DEBUG_MODE == 'detect_nan':  # use suggested Theano detect_nan implementation
+        mode = theano.compile.MonitorMode(post_func=theano.compile.monitormode.detect_nan).excluding('local_elemwise_fusion', 'inplace')
```

Comment:
Personnaly, I would use Theano flags :) I don't know if it should be added directly in Theano. Do tensorflow support this? If it is added in keras, I would do:
```
mode = None
if _DEBUG_MODE == 'detect_nan':
    mode = 'NanGuardMode'
```

Can you open an issue on Theano, Using the Theano flag should not enable the GPU. I don't have this behavior on the lstm example.